Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -433,20 +433,40 @@ public override void Update(float frameTime)
 
                 if (flammable.FireStacks > 0)
                 {
-                    var air = _atmosphereSystem.GetContainingMixture(uid);
+                    var air = _atmosphereSystem.GetContainingMixture(uid, false, true);
 
-                    // If we're in an oxygenless environment, put the fire out.
-                    if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                    if (flammable.RequireOxygen)
                     {
-                        Extinguish(uid, flammable);
-                        continue;
+                        // If we're in an oxygenless environment, put the fire out.
+                        if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                        {
+                            Extinguish(uid, flammable);
+                            continue;
+                        }
                     }
 
                     var source = EnsureComp<IgnitionSourceComponent>(uid);
                     _ignitionSourceSystem.SetIgnited((uid, source));
 
                     if (TryComp(uid, out TemperatureComponent? temp))
-                        _temperatureSystem.ChangeHeat(uid, 12500 * flammable.FireStacks, false, temp);
+                    {
+                        // get hot
+                        var maxDeltaT = flammable.PeakFlameTemperature - temp.CurrentTemperature;
```

Comment:
This doesn't really make sense to me. When something burns, it adds energy in the form of heat. Why should there be a `PeakFlameTemperature` cap?

Reply:
Real things don't just gen infinitely hotter as they burn, they reach an equilibrium. Also this led to crazy things like 22000k candles by the time they burnt up, and I'm 90% sure was also causing a lot of the issues with people ashing constantly from small fires.

Reply:
In a real system, this equilibrium is reached when the rate of energy removed (e.g. by convective cooling) equals the rate of energy added. If you had a closed system and infinite fuel, then yes it would get infinitely hotter. This is just conservation of energy.

The real problem is that we have superconduction off -- but we should fix it there and not here.

Reply:
![image](https://github.com/user-attachments/assets/14f6b630-f58e-49b6-abd7-3c115f5e4fd7)

Not really sure where to go from here, it certainly wont work just leaving the temperature as a runaway though.

Reply:
The candle isn't infinte, perhaps add a "energy added per second" datafield to the candle and add that much energy per second.

Reply:
This also effects things that might not destruct though, and those will run away. I'm not much of a physicist, but I think what we're getting at is the [Adiabatic Flame Temperature](https://en.wikipedia.org/wiki/Adiabatic_flame_temperature). If we were using the superconducting it would happen 'naturally' but since we're not (and since we don't actually consume the gas as fuel from the tile air mix while entities burn, but that's a different confounding issue) we need to fake the temperature asymptoting to prevent things from just getting ever-hotter.

note: That wiki article mentions both constant pressure and constant volume, I think the constant volume one matches our use case the most since the tile volumes don't change.

Reply:
Okay I looked at this again and I think this was a bit of a red hearring. I don't think we need superconduction at all here, and I actually dont think it would have solved this problem. Instead I went with making flammable entities have the AtmosExposedComponent, and also made it so the AtmosExposed component preserves energy between entities and the atmosphere by actually moving the energy and not just creating it out of thin air (ha!).